utils/UUID_gen.hh

```diff
@@ -383,7 +383,11 @@ public:
         auto dmc = duration_cast<decimicroseconds>(d);
         uint64_t msb = dmc.count();
         // timeuuid time must fit in 60 bits
-        assert(!(0xf000000000000000UL & msb));
+        if ((0xf000000000000000UL & msb)) {
+            // We hope callers would try to avoid this case, but they don't
+            // always do, so assert() would be bad here - and caused #17035.
+            throw std::runtime_error("timeuuid time must fit in 60 bits");
```

Comment:
Let's also include the bad input, so when we see it in the logs, we don't have to guess.

Reply:
Ok, I'll do that.

I didn't do this because 1. it was a pre-existing message, and 2. it's a header file and doesn't yet use fmt :-(
But I guess every source file in Scylla already uses fmt so adding it here won't hurt issue #1 because it won't really be included to any source file that doesn't already include it..

Reply:
Oh second thought, I'm not sure what it will help to print the input. It's usually already converted from some other format by the caller, and the problem in the after-conversion input is obvious - it has high-order bits on, we don't need to print it to see it. Or maybe you think seeing "msb" in full will be helpful?

What I think might be more useful is a backtrace, exactly what on_internal_error() does.
Yesterday I actually wanted to use on_internal_error here when I wrote this patch, but couldn't because it's a header file and I don't have a logger. This is why I used throw :-( Today I saw you also needed on_internal_error in a header file, so I think it's time to solve this problem. What do you say about me adding a utils::on_internal_error which has its own logger, and I'll use it here in the header file?

Reply:
I'm onboard. I think in general we should replace `assert()` with `on_internal_error()`.